jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/RemoveDataCaseFileInstanceCommand.java

```diff
@@ -40,9 +42,11 @@
 
     private List<String> variableNames;
     private AuthorizationManager authorizationManager;
+    private Long processInstanceId;
     
-    public RemoveDataCaseFileInstanceCommand(IdentityProvider identityProvider, List<String> variableNames, AuthorizationManager authorizationManager) {                
+    public RemoveDataCaseFileInstanceCommand(Long processInstanceId, IdentityProvider identityProvider, List<String> variableNames, AuthorizationManager authorizationManager) {                
```

Comment:
In theory somebody can be already using that via `ProcessService` to be run as a separate command, so we can break backward compatibility, but if there is no other way, it is ok from my side.

Reply:
Im aware of the potential backward compatiiliry problem, but I cannot find other way, processInstanceId is needed to check the readonly flag (and now both command constructors, add and remove data case are symetrical)